### PR TITLE
Pass child initiated event version to child workflow

### PR DESCRIPTION
**Pull request by yycptt**
**What changed?**
- Pass child initiated event version (in parent workflow) to child workflow
- API change: https://github.com/temporalio/api/pull/180
- Renamed initiated_id to parent_initiated_id in WorkflowExecutionInfo and RecordChildExecutionCompleted

**Why?**
- When child respond back to parent, it needs the version + eventID to uniquely identify a child init event in parent in xdc case.

**How did you test it?**
- Modified existing tests

**Potential risks**
-  N/A only adding the field. No logic is using the new fields.

**Is hotfix candidate?**
- No.

### CODE REVIEW

File: proto/internal/temporal/server/api/persistence/v1/executions.proto [hunk 1]
```diff
@@ -130,6 +130,8 @@ message WorkflowExecutionInfo {
     // If continued-as-new, or retried, or cron, holds the new run id.
     string new_execution_run_id = 61;
     temporal.server.api.clock.v1.ShardClock parent_clock = 62;
+    // version of child execution initiated event in parent workflow
+    int64 initiated_version = 63;
```
Comment (wxing1292) [inline proto/internal/temporal/server/api/persistence/v1/executions.proto:134]: rename `int64 initiated_id = 6;` -> `int64 parent_initiated_id = 6;`
and this `int64 initiated_version = 63; ` -> `int64 parent_initiated_version = 63; `

?

File: service/history/workflow/mutable_state_impl.go [hunk 2]
```diff
@@ -1544,12 +1544,8 @@ func (e *MutableStateImpl) ReplicateWorkflowExecutionStartedEvent(
...
 		e.executionInfo.ParentRunId = event.ParentWorkflowExecution.GetRunId()
 		e.executionInfo.ParentClock = parentClock
-	}
-
-	if event.ParentInitiatedEventId != 0 {
 		e.executionInfo.InitiatedId = event.GetParentInitiatedEventId()
-	} else {
-		e.executionInfo.InitiatedId = common.EmptyEventID
+		e.executionInfo.InitiatedVersion = event.GetParentInitiatedEventVersion()
 	}
```
Comment (wxing1292) [inline service/history/workflow/mutable_state_impl.go:1549]: can we keep the else block?
```go
e.executionInfo.InitiatedId = common.EmptyEventID
e.executionInfo. InitiatedVersion = common.EmptyVersion
```

**Review by wxing1292 — APPROVED** [review]

File: service/history/workflow/mutable_state_impl.go (hunk 2 above)
Comment (yycptt) [reply service/history/workflow/mutable_state_impl.go]: Added back the check. also added a new if block to check the initiated version.

File: proto/internal/temporal/server/api/persistence/v1/executions.proto (hunk 1 above)
Comment (yycptt) [reply proto/internal/temporal/server/api/persistence/v1/executions.proto]: Update the naming in WorkflowExecutionInfo and RecordChildExecutionCompleted.

**Pull request merged**